phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexScrutinyMapper.java

```diff
@@ -151,6 +151,16 @@ protected void setup(final Context context) throws IOException, InterruptedExcep
             LOG.info("Target table base query: " + targetTableQuery);
             md5 = MessageDigest.getInstance("MD5");
         } catch (SQLException | NoSuchAlgorithmException e) {
+            if (this.connection != null) {
+                try {
+                    this.connection.close();
```

Comment:
in the rare case that this throws, `outputConn` will not get closed.  Looks like we need to fix it in `cleanup()` also